Examples/Algorithms/TrackFinding/src/HoughTransformSeeder.cpp

```diff
@@ -0,0 +1,529 @@
+              double r = sqrt(sp->x() * sp->x() + sp->y() * sp->y());
+              unsigned layer = (*(m_cfg.findLayerIDSP))(r);
+              hitIndicesAll[layer].push_back(sp->measurementIndex());
+              nHitsPerLayer[layer]++;
+            } else {  // we store SP first, this must not be a sp, updated the
+                      // index appropriately
+              std::shared_ptr<const MeasurementStruct> meas =
+                  measurementStructs[index - spacePointPtrs.size()];
+              unsigned layer = meas->layer;
+              hitIndicesAll[layer].push_back(meas->index);
+              nHitsPerLayer[layer]++;
+            }
+          }
+
+          std::vector<std::vector<int>> combs = getComboIndices(nHitsPerLayer);
+
+          for (size_t icomb = 0; icomb < combs.size();
+               icomb++) {  // loop over all the combinations
+            ProtoTrack protoTrack;
+            std::vector<int> const& hit_indices =
+                combs[icomb];  // one index per layer
+            for (unsigned layer = 0; layer < m_cfg.m_nLayers; layer++) {
+              if (hit_indices[layer] >= 0) {
+                protoTrack.push_back(hitIndicesAll[layer][hit_indices[layer]]);
+              }
+            }
+            protoTracks.push_back(protoTrack);
+          }
+        }
+  }
+  ACTS_DEBUG("Created " << protoTracks.size() << " track seeds from "
+                        << spacePointPtrs.size() << " space points");
+
+  ctx.eventStore.add(m_cfg.outputProtoTracks, ProtoTrackContainer{protoTracks});
+
+  return ActsExamples::ProcessCode::SUCCESS;
+}
+
+ActsExamples::Image ActsExamples::HoughTransformSeeder::createLayerImage(
+    unsigned layer, std::vector<const SimSpacePoint*>& spacepoints,
+    std::vector<std::shared_ptr<const MeasurementStruct>>& meass,
+    int subregion) const {
+  ActsExamples::Image image(m_cfg.m_imageSize_y, m_cfg.m_imageSize_x);
+
+  for (unsigned index = 0; index < spacepoints.size(); index++) {
+    const SimSpacePoint* sp = spacepoints[index];
+    double r = sqrt(sp->x() * sp->x() + sp->y() * sp->y());
+    double z = sp->z();
+    unsigned hitlayer = (*(m_cfg.findLayerIDSP))(r);
+    if (hitlayer != layer)
+      continue;
+    if (!((*(m_cfg.inSliceSP))(z, hitlayer, subregion)))
+      continue;
+    float phi = atan2(sp->y(), sp->x());
+
+    // This scans over y (pT) because that is more efficient in memory, in C.
+    // Unknown if firmware will want to scan over x instead.
+    for (unsigned y_ = 0; y_ < m_cfg.m_imageSize_y; y_++) {
+      unsigned y_bin_min = y_;
+      unsigned y_bin_max = (y_ + 1);
+
+      // Find the min/max x bins
+      auto xBins = yToXBins(y_bin_min, y_bin_max, r, phi, hitlayer);
+      // Update the image
+      for (unsigned y = y_bin_min; y < y_bin_max; y++)
+        for (unsigned x = xBins.first; x < xBins.second; x++) {
+          image(y, x).first++;
+          image(y, x).second.insert(index);
```

Comment:
this... adds... each spacepoint of the current layer to EVERY x AND y bin of this layerImage... no?

Reply:
No, see our previous discussion